prow/spyglass/artifacts.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"context"
+	"io/ioutil"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+)
+
+// GCSArtifact represents some output of a prow job stored in GCS
+type GCSArtifact struct {
+	Artifact
+
+	// The handle of the object in GCS
+	Handle storage.ObjectHandle
+
+	// The direct link to the Artifact, can be used for read operations
+	link string
+
+	// The path of the Artifact within the job
+	path string
+}
+
+// Gets the GCS path of the artifact within the current job
+func (a GCSArtifact) JobPath() string {
+	return a.path
+}
+
+// Gets the GCS web address of the artifact
+func (a GCSArtifact) CanonicalLink() string {
+	return a.link
+}
+
+// Reads len(p) bytes from a file in GCS
+func (a GCSArtifact) Read(p []byte) (n int, err error) {
```

Comment:
the signature is not consistent between `Read` and `ReadAll`, I think this should take in n, and return []byte, error

Reply:
The signatures are intentionally different. This signature is taken from the Go standard io library. The `ReadAll` signature is taken from the Go standard io/ioutil library.